scripts/ci-entrypoint.sh

```diff
@@ -36,35 +36,14 @@ source "${REPO_ROOT}/hack/ensure-kustomize.sh"
 # shellcheck source=../hack/parse-prow-creds.sh
 source "${REPO_ROOT}/hack/parse-prow-creds.sh"
 
-# build Kubernetes E2E binaries
-build_k8s() {
-    # possibly enable bazel build caching before building kubernetes
-    if [[ "${BAZEL_REMOTE_CACHE_ENABLED:-false}" == "true" ]]; then
-        create_bazel_cache_rcs.sh || true
-    fi
-
-    pushd "$(go env GOPATH)/src/k8s.io/kubernetes"
-
-    # make sure we have e2e requirements
-    bazel build //cmd/kubectl //test/e2e:e2e.test //vendor/github.com/onsi/ginkgo/ginkgo
-
-    # ensure the e2e script will find our binaries ...
-    mkdir -p "${PWD}/_output/bin/"
-    cp -f "${PWD}/bazel-bin/test/e2e/e2e.test" "${PWD}/_output/bin/e2e.test"
-
-    PATH="$(dirname "$(find "${PWD}/bazel-bin/" -name kubectl -type f)"):${PATH}"
-    export PATH
-
-    # attempt to release some memory after building
-    sync || true
-    (echo 1 > /proc/sys/vm/drop_caches) 2>/dev/null || true
-
-    popd
-}
-
 create_cluster() {
-    # export cluster template which contains the manifests needed for creating the Azure cluster to run the tests
-    if [[ -n ${CI_VERSION:-} || -n ${USE_CI_ARTIFACTS:-} ]]; then
+    # JOB_TYPE is an enviornment variable set by a prow job -
+    # https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables
+    if [[ "${JOB_TYPE:-}" == "presubmit" ]]; then
```

Comment:
so this makes sense in the context of a presubmit job running against k/k but what about the jobs that run in capz? do we want to be using a custom build of k8s for those?

Reply:
Probably doesn't make sense to build Kubernetes for PRs in cluster-api-provider-azure repository. We probably need an additional check/environment variable to control that.